app/screens/custom_status_clear_after/date_time_selector.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import moment, {Moment} from 'moment-timezone';
+import React, {useState} from 'react';
+import {View, Button, Platform} from 'react-native';
+import {useSelector} from 'react-redux';
+
+import Preferences from '@mm-redux/constants/preferences';
+import {getBool} from '@mm-redux/selectors/entities/preferences';
+import {getCurrentUserTimezone} from '@mm-redux/selectors/entities/timezone';
+import {Theme} from '@mm-redux/types/preferences';
+import {GlobalState} from '@mm-redux/types/store';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {getCurrentMomentForTimezone, getUtcOffsetForTimeZone} from '@utils/timezone';
+
+type Props = {
+    theme: Theme;
+    handleChange: (currentDate: Moment) => void;
+}
+
+type AndroidMode = 'date' | 'time';
+
+const CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES = 30;
+export function getRoundedTime(value: Moment) {
+    const roundedTo = CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES;
+    const start = moment(value);
+    const diff = start.minute() % roundedTo;
+    if (diff === 0) {
+        return value;
+    }
+    const remainder = roundedTo - diff;
+    return start.add(remainder, 'm').seconds(0).milliseconds(0);
+}
+
+const DateTimeSelector = (props: Props) => {
+    const {theme} = props;
+    const styles = getStyleSheet(theme);
+    const militaryTime = useSelector((state: GlobalState) => getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, 'use_military_time'));
+    const timezone = useSelector(getCurrentUserTimezone);
+    const currentTime = getCurrentMomentForTimezone(timezone);
+    const timezoneOffSetInMinutes = timezone ? getUtcOffsetForTimeZone(timezone) : undefined;
+
+    const minimumDate = getRoundedTime(currentTime);
+
+    const [date, setDate] = useState<Moment>(minimumDate);
+    const [mode, setMode] = useState<AndroidMode>('date');
+    const [show, setShow] = useState<boolean>(false);
+
+    const onChange = (_: React.ChangeEvent<HTMLInputElement>, selectedDate: Date) => {
+        const currentDate = selectedDate || date;
+        setShow(Platform.OS === 'ios');
+        if (moment(currentDate).isAfter(minimumDate)) {
```

Comment:
This fixes the bug which is great. I'd leave the UX to someone else, wonder if feedback is needed to avoid frustration of tapping on an option to select it and nothing happens. @esethna thoughts? Matt is out so up to you 🤟